WordPress-iOS-Shared/Core/WPImageSource.m

```diff
@@ -24,17 +25,19 @@ + (instancetype)sharedSource
 
 - (void)dealloc
 {
-    [_downloadingQueue cancelAllOperations];
+    [_downloadsSession invalidateAndCancel];
 }
 
 - (id)init
 {
     self = [super init];
     if (self) {
-        _downloadingQueue = [[NSOperationQueue alloc] init];
         _urlDownloadsInProgress = [[NSMutableSet alloc] init];
         _successBlocks = [[NSMutableDictionary alloc] init];
         _failureBlocks = [[NSMutableDictionary alloc] init];
+
+        NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+        _downloadsSession = [NSURLSession sessionWithConfiguration:configuration];
```

Comment:
Thank you for the answering for me @koke I agree that background downloads are not desirable here. And even if we wanted them will not be easy to implement because then you cannot use the session block interface, and you need to complete callbacks for background downloads if the the app is not running.